cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -224,6 +235,23 @@ protected NewTopic createMetricsTopicFromReporterConfig(CruiseControlMetricsRepo
     return newTopic;
   }
 
+  protected void createCruiseControlMetricsProducer(Properties producerProps) throws KafkaException {
+    CruiseControlMetricsUtils.retry(() -> {
+      try {
+        _producer = new KafkaProducer<>(producerProps);
+        return false;
+      } catch (KafkaException e) {
+        if (e.getCause() instanceof ConfigException
+                && e.getCause().toString().contains("No resolvable bootstrap urls given in bootstrap.servers")) {
```

Comment:
Instead of checking for a specific string isn't there a specific type of exception that would indicate the connection failure?

Reply:
The exception is called `ConfigException` and it was kind of vague, so that's why I was checking the message too. I can remove the message check if you prefer.